ICE: `has escaping bound vars, so it cannot be wrapped in a dummy binder.`

<!--
Thank you for finding an Internal Compiler Error! 🧊  If possible, try to provide
a minimal verifiable example. You can read "Rust Bug Minimization Patterns" for
how to create smaller examples.
http://blog.pnkfx.org/blog/2019/11/18/rust-bug-minimization-patterns/
-->

### Code

```Rust
mod assert {
    use std::mem::{Assume, BikeshedIntrinsicFrom};

    pub fn is_maybe_transmutable<Src, Dst>()
    where
        Dst: BikeshedIntrinsicFrom<
            Src,
            Context,
            { Assume::ALIGNMENT.and().and(<x as Fn(&usize)>::call).and() },
        >,
    {
    }
}
```


### Meta
<!--
If you're using the stable version of the compiler, you should also check if the
bug also exists in the beta or nightly versions.
-->

`rustc --version --verbose`:
```
rustc 1.74.0-nightly (9f5fc1bd4 2023-09-02)
binary: rustc
commit-hash: 9f5fc1bd443f59583e7af0d94d289f95fe1e20c4
commit-date: 2023-09-02
host: x86_64-unknown-linux-gnu
release: 1.74.0-nightly
LLVM version: 17.0.0
```



<details><summary><strong>Error output</strong></summary>
<p>

```
error[E0412]: cannot find type `Context` in this scope
 --> AE0486D4D1D9D2A1A6C2A038AAD306BAF806AFB187CBB65F18D06E9837A46E84.rs:8:13
  |
8 |             Context,
  |             ^^^^^^^ not found in this scope
  |
help: consider importing this struct
  |
2 +     use std::task::Context;
  |

error[E0412]: cannot find type `x` in this scope
 --> AE0486D4D1D9D2A1A6C2A038AAD306BAF806AFB187CBB65F18D06E9837A46E84.rs:9:44
  |
9 |             { Assume::ALIGNMENT.and().and(<x as Fn(&usize)>::call).and() },
  |                                            ^ not found in this scope
  |
help: you might be missing a type parameter
  |
4 |     pub fn is_maybe_transmutable<Src, Dst, x>()
  |                                          +++

error[E0601]: `main` function not found in crate `AE0486D4D1D9D2A1A6C2A038AAD306BAF806AFB187CBB65F18D06E9837A46E84`
  --> AE0486D4D1D9D2A1A6C2A038AAD306BAF806AFB187CBB65F18D06E9837A46E84.rs:13:2
   |
13 | }
   |  ^ consider adding a `main` function to `AE0486D4D1D9D2A1A6C2A038AAD306BAF806AFB187CBB65F18D06E9837A46E84.rs`

error[E0658]: use of unstable library feature 'transmutability'
 --> AE0486D4D1D9D2A1A6C2A038AAD306BAF806AFB187CBB65F18D06E9837A46E84.rs:2:20
  |
2 |     use std::mem::{Assume, BikeshedIntrinsicFrom};
  |                    ^^^^^^
  |
  = note: see issue #99571 <https://github.com/rust-lang/rust/issues/99571> for more information
  = help: add `#![feature(transmutability)]` to the crate attributes to enable

error[E0658]: use of unstable library feature 'transmutability'
 --> AE0486D4D1D9D2A1A6C2A038AAD306BAF806AFB187CBB65F18D06E9837A46E84.rs:2:28
  |
2 |     use std::mem::{Assume, BikeshedIntrinsicFrom};
  |                            ^^^^^^^^^^^^^^^^^^^^^
  |
  = note: see issue #99571 <https://github.com/rust-lang/rust/issues/99571> for more information
  = help: add `#![feature(transmutability)]` to the crate attributes to enable

error[E0658]: use of unstable library feature 'transmutability'
  --> AE0486D4D1D9D2A1A6C2A038AAD306BAF806AFB187CBB65F18D06E9837A46E84.rs:6:14
   |
6  |           Dst: BikeshedIntrinsicFrom<
   |  ______________^
7  | |             Src,
8  | |             Context,
9  | |             { Assume::ALIGNMENT.and().and(<x as Fn(&usize)>::call).and() },
10 | |         >,
   | |_________^
   |
   = note: see issue #99571 <https://github.com/rust-lang/rust/issues/99571> for more information
   = help: add `#![feature(transmutability)]` to the crate attributes to enable

error[E0658]: use of unstable library feature 'transmutability'
 --> AE0486D4D1D9D2A1A6C2A038AAD306BAF806AFB187CBB65F18D06E9837A46E84.rs:9:15
  |
9 |             { Assume::ALIGNMENT.and().and(<x as Fn(&usize)>::call).and() },
  |               ^^^^^^
  |
  = note: see issue #99571 <https://github.com/rust-lang/rust/issues/99571> for more information
  = help: add `#![feature(transmutability)]` to the crate attributes to enable

error[E0658]: use of unstable library feature 'fn_traits'
 --> AE0486D4D1D9D2A1A6C2A038AAD306BAF806AFB187CBB65F18D06E9837A46E84.rs:9:43
  |
9 |             { Assume::ALIGNMENT.and().and(<x as Fn(&usize)>::call).and() },
  |                                           ^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: see issue #29625 <https://github.com/rust-lang/rust/issues/29625> for more information
  = help: add `#![feature(fn_traits)]` to the crate attributes to enable

error[E0658]: use of unstable library feature 'transmutability'
 --> AE0486D4D1D9D2A1A6C2A038AAD306BAF806AFB187CBB65F18D06E9837A46E84.rs:9:15
  |
9 |             { Assume::ALIGNMENT.and().and(<x as Fn(&usize)>::call).and() },
  |               ^^^^^^^^^^^^^^^^^
  |
  = note: see issue #99571 <https://github.com/rust-lang/rust/issues/99571> for more information
  = help: add `#![feature(transmutability)]` to the crate attributes to enable

error[E0658]: use of unstable library feature 'transmutability'
 --> AE0486D4D1D9D2A1A6C2A038AAD306BAF806AFB187CBB65F18D06E9837A46E84.rs:9:33
  |
9 |             { Assume::ALIGNMENT.and().and(<x as Fn(&usize)>::call).and() },
  |                                 ^^^
  |
  = note: see issue #99571 <https://github.com/rust-lang/rust/issues/99571> for more information
  = help: add `#![feature(transmutability)]` to the crate attributes to enable

error[E0061]: this method takes 1 argument but 0 arguments were supplied
  --> AE0486D4D1D9D2A1A6C2A038AAD306BAF806AFB187CBB65F18D06E9837A46E84.rs:9:33
   |
9  |             { Assume::ALIGNMENT.and().and(<x as Fn(&usize)>::call).and() },
   |                                 ^^^-- an argument of type `Assume` is missing
   |
note: method defined here
  --> /home/matthias/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/mem/transmutability.rs:67:18
   |
67 |     pub const fn and(self, other_assumptions: Self) -> Self {
   |                  ^^^
help: provide the argument
   |
9  |             { Assume::ALIGNMENT.and(/* Assume */).and(<x as Fn(&usize)>::call).and() },
   |                                    ~~~~~~~~~~~~~~

error[E0658]: use of unstable library feature 'transmutability'
 --> AE0486D4D1D9D2A1A6C2A038AAD306BAF806AFB187CBB65F18D06E9837A46E84.rs:9:39
  |
9 |             { Assume::ALIGNMENT.and().and(<x as Fn(&usize)>::call).and() },
  |                                       ^^^
  |
  = note: see issue #99571 <https://github.com/rust-lang/rust/issues/99571> for more information
  = help: add `#![feature(transmutability)]` to the crate attributes to enable

error[E0229]: associated type bindings are not allowed here
 --> AE0486D4D1D9D2A1A6C2A038AAD306BAF806AFB187CBB65F18D06E9837A46E84.rs:9:49
  |
9 |             { Assume::ALIGNMENT.and().and(<x as Fn(&usize)>::call).and() },
  |                                                 ^^^^^^^^^^ associated type not allowed here
```

</p>
</details>

<!--
Include a backtrace in the code block by setting `RUST_BACKTRACE=1` in your
environment. E.g. `RUST_BACKTRACE=1 cargo build`.
-->
<details><summary><strong>Backtrace</strong></summary>
<p>

```

thread 'rustc' panicked at compiler/rustc_hir_typeck/src/fn_ctxt/_impl.rs:480:13:
`WellFormed((&usize,))` has escaping bound vars, so it cannot be wrapped in a dummy binder.
stack backtrace:
   0:     0x7f9c4dcdbdec - std::backtrace_rs::backtrace::libunwind::trace::hdcb9c0b22500da91
                               at /rustc/9f5fc1bd443f59583e7af0d94d289f95fe1e20c4/library/std/src/../../backtrace/src/backtrace/libunwind.rs:93:5
   1:     0x7f9c4dcdbdec - std::backtrace_rs::backtrace::trace_unsynchronized::h91b351fd34050178
                               at /rustc/9f5fc1bd443f59583e7af0d94d289f95fe1e20c4/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5
   2:     0x7f9c4dcdbdec - std::sys_common::backtrace::_print_fmt::he469089ab3fbd8a5
                               at /rustc/9f5fc1bd443f59583e7af0d94d289f95fe1e20c4/library/std/src/sys_common/backtrace.rs:67:5
   3:     0x7f9c4dcdbdec - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h57d5e31c47a8a531
                               at /rustc/9f5fc1bd443f59583e7af0d94d289f95fe1e20c4/library/std/src/sys_common/backtrace.rs:44:22
   4:     0x7f9c4dd4205c - core::fmt::rt::Argument::fmt::h6e8bafe434c307b3
                               at /rustc/9f5fc1bd443f59583e7af0d94d289f95fe1e20c4/library/core/src/fmt/rt.rs:138:9
   5:     0x7f9c4dd4205c - core::fmt::write::hbf1ef8babec800ef
                               at /rustc/9f5fc1bd443f59583e7af0d94d289f95fe1e20c4/library/core/src/fmt/mod.rs:1094:21
   6:     0x7f9c4dcce8fe - std::io::Write::write_fmt::h3273b2319fbd2ddc
                               at /rustc/9f5fc1bd443f59583e7af0d94d289f95fe1e20c4/library/std/src/io/mod.rs:1714:15
   7:     0x7f9c4dcdbbd4 - std::sys_common::backtrace::_print::h9f429c8b9622817f
                               at /rustc/9f5fc1bd443f59583e7af0d94d289f95fe1e20c4/library/std/src/sys_common/backtrace.rs:47:5
   8:     0x7f9c4dcdbbd4 - std::sys_common::backtrace::print::h3e62e3dc9cc0960c
                               at /rustc/9f5fc1bd443f59583e7af0d94d289f95fe1e20c4/library/std/src/sys_common/backtrace.rs:34:9
   9:     0x7f9c4dcdecea - std::panicking::panic_hook_with_disk_dump::{{closure}}::h336ab5ed5f752f5b
                               at /rustc/9f5fc1bd443f59583e7af0d94d289f95fe1e20c4/library/std/src/panicking.rs:280:22
  10:     0x7f9c4dcde9e5 - std::panicking::panic_hook_with_disk_dump::h0b2ed72036e73f37
                               at /rustc/9f5fc1bd443f59583e7af0d94d289f95fe1e20c4/library/std/src/panicking.rs:314:9
  11:     0x7f9c4cb4d079 - rustc_driver_impl[bd3214347f952018]::install_ice_hook::{closure#0}
  12:     0x7f9c4dcdf5a3 - <alloc::boxed::Box<F,A> as core::ops::function::Fn<Args>>::call::hf5daf4aeb8dae1f0
                               at /rustc/9f5fc1bd443f59583e7af0d94d289f95fe1e20c4/library/alloc/src/boxed.rs:2021:9
  13:     0x7f9c4dcdf5a3 - std::panicking::rust_panic_with_hook::h725e440d3f428ec3
                               at /rustc/9f5fc1bd443f59583e7af0d94d289f95fe1e20c4/library/std/src/panicking.rs:757:13
  14:     0x7f9c4dcdf321 - std::panicking::begin_panic_handler::{{closure}}::h68f67ead3656c6d6
                               at /rustc/9f5fc1bd443f59583e7af0d94d289f95fe1e20c4/library/std/src/panicking.rs:631:13
  15:     0x7f9c4dcdc316 - std::sys_common::backtrace::__rust_end_short_backtrace::h92cd6f62fea1143e
                               at /rustc/9f5fc1bd443f59583e7af0d94d289f95fe1e20c4/library/std/src/sys_common/backtrace.rs:170:18
  16:     0x7f9c4dcdf062 - rust_begin_unwind
                               at /rustc/9f5fc1bd443f59583e7af0d94d289f95fe1e20c4/library/std/src/panicking.rs:619:5
  17:     0x7f9c4dd3e405 - core::panicking::panic_fmt::he6a20279c6d1da12
                               at /rustc/9f5fc1bd443f59583e7af0d94d289f95fe1e20c4/library/core/src/panicking.rs:72:14
  18:     0x7f9c4aeb2504 - <rustc_hir_typeck[2c4653fa0fac3fb]::fn_ctxt::FnCtxt>::register_wf_obligation
  19:     0x7f9c4aeb2285 - <rustc_hir_typeck[2c4653fa0fac3fb]::fn_ctxt::FnCtxt>::to_ty
  20:     0x7f9c4af24207 - <<rustc_hir_typeck[2c4653fa0fac3fb]::fn_ctxt::FnCtxt>::instantiate_value_path::CreateCtorSubstsContext as rustc_hir_analysis[f4190e851159bd14]::astconv::CreateSubstsForGenericArgsCtxt>::provided_kind
  21:     0x7f9c4aed954b - <rustc_hir_typeck[2c4653fa0fac3fb]::fn_ctxt::FnCtxt>::check_expr_path
  22:     0x7f9c4ae9af8f - <rustc_hir_typeck[2c4653fa0fac3fb]::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args
  23:     0x7f9c4aef672d - <rustc_hir_typeck[2c4653fa0fac3fb]::fn_ctxt::FnCtxt>::check_argument_types
  24:     0x7f9c4ae9cb6d - <rustc_hir_typeck[2c4653fa0fac3fb]::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args
  25:     0x7f9c4ae9bdb4 - <rustc_hir_typeck[2c4653fa0fac3fb]::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args
  26:     0x7f9c4aecdacd - <rustc_hir_typeck[2c4653fa0fac3fb]::fn_ctxt::FnCtxt>::check_block_with_expected
  27:     0x7f9c4ae9bd5d - <rustc_hir_typeck[2c4653fa0fac3fb]::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args
  28:     0x7f9c4ae97d36 - <rustc_hir_typeck[2c4653fa0fac3fb]::fn_ctxt::FnCtxt>::check_expr_coercible_to_type
  29:     0x7f9c4bb766d8 - rustc_hir_typeck[2c4653fa0fac3fb]::typeck
  30:     0x7f9c4ac57cae - rustc_query_impl[2b96a7f183453ea1]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[2b96a7f183453ea1]::query_impl::typeck::dynamic_query::{closure#2}::{closure#0}, rustc_middle[a400c0366bd3264d]::query::erase::Erased<[u8; 8usize]>>
  31:     0x7f9c4ac57c7e - <rustc_query_impl[2b96a7f183453ea1]::query_impl::typeck::dynamic_query::{closure#2} as core[afd64c1140d34ddb]::ops::function::FnOnce<(rustc_middle[a400c0366bd3264d]::ty::context::TyCtxt, rustc_span[d6219d1cb1a2cd02]::def_id::LocalDefId)>>::call_once
  32:     0x7f9c4ad49222 - rustc_query_system[f349c9744a75bab0]::query::plumbing::try_execute_query::<rustc_query_impl[2b96a7f183453ea1]::DynamicConfig<rustc_query_system[f349c9744a75bab0]::query::caches::VecCache<rustc_span[d6219d1cb1a2cd02]::def_id::LocalDefId, rustc_middle[a400c0366bd3264d]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[2b96a7f183453ea1]::plumbing::QueryCtxt, false>
  33:     0x7f9c4c4e9901 - rustc_query_impl[2b96a7f183453ea1]::query_impl::typeck::get_query_non_incr::__rust_end_short_backtrace
  34:     0x7f9c4b506099 - rustc_mir_build[a29c2658498d3592]::thir::cx::thir_body
  35:     0x7f9c4ac238d1 - rustc_query_impl[2b96a7f183453ea1]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[2b96a7f183453ea1]::query_impl::thir_body::dynamic_query::{closure#2}::{closure#0}, rustc_middle[a400c0366bd3264d]::query::erase::Erased<[u8; 16usize]>>
  36:     0x7f9c4b224211 - rustc_query_system[f349c9744a75bab0]::query::plumbing::try_execute_query::<rustc_query_impl[2b96a7f183453ea1]::DynamicConfig<rustc_query_system[f349c9744a75bab0]::query::caches::VecCache<rustc_span[d6219d1cb1a2cd02]::def_id::LocalDefId, rustc_middle[a400c0366bd3264d]::query::erase::Erased<[u8; 16usize]>>, false, false, false>, rustc_query_impl[2b96a7f183453ea1]::plumbing::QueryCtxt, false>
  37:     0x7f9c4c4df064 - rustc_query_impl[2b96a7f183453ea1]::query_impl::thir_body::get_query_non_incr::__rust_end_short_backtrace
  38:     0x7f9c4c383a4b - rustc_mir_build[a29c2658498d3592]::thir::pattern::check_match::check_match
  39:     0x7f9c4ac0a17c - rustc_query_impl[2b96a7f183453ea1]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[2b96a7f183453ea1]::query_impl::check_match::dynamic_query::{closure#2}::{closure#0}, rustc_middle[a400c0366bd3264d]::query::erase::Erased<[u8; 1usize]>>
  40:     0x7f9c4b40a116 - rustc_query_system[f349c9744a75bab0]::query::plumbing::try_execute_query::<rustc_query_impl[2b96a7f183453ea1]::DynamicConfig<rustc_query_system[f349c9744a75bab0]::query::caches::VecCache<rustc_span[d6219d1cb1a2cd02]::def_id::LocalDefId, rustc_middle[a400c0366bd3264d]::query::erase::Erased<[u8; 1usize]>>, false, false, false>, rustc_query_impl[2b96a7f183453ea1]::plumbing::QueryCtxt, false>
  41:     0x7f9c4b409df4 - rustc_query_impl[2b96a7f183453ea1]::query_impl::check_match::get_query_non_incr::__rust_end_short_backtrace
  42:     0x7f9c4ba03e9d - rustc_mir_build[a29c2658498d3592]::build::mir_built
  43:     0x7f9c4ac2ce8c - rustc_query_impl[2b96a7f183453ea1]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[2b96a7f183453ea1]::query_impl::mir_built::dynamic_query::{closure#2}::{closure#0}, rustc_middle[a400c0366bd3264d]::query::erase::Erased<[u8; 8usize]>>
  44:     0x7f9c4ac2ce6e - <rustc_query_impl[2b96a7f183453ea1]::query_impl::mir_built::dynamic_query::{closure#2} as core[afd64c1140d34ddb]::ops::function::FnOnce<(rustc_middle[a400c0366bd3264d]::ty::context::TyCtxt, rustc_span[d6219d1cb1a2cd02]::def_id::LocalDefId)>>::call_once
  45:     0x7f9c4ad49222 - rustc_query_system[f349c9744a75bab0]::query::plumbing::try_execute_query::<rustc_query_impl[2b96a7f183453ea1]::DynamicConfig<rustc_query_system[f349c9744a75bab0]::query::caches::VecCache<rustc_span[d6219d1cb1a2cd02]::def_id::LocalDefId, rustc_middle[a400c0366bd3264d]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[2b96a7f183453ea1]::plumbing::QueryCtxt, false>
  46:     0x7f9c4c4df731 - rustc_query_impl[2b96a7f183453ea1]::query_impl::mir_built::get_query_non_incr::__rust_end_short_backtrace
  47:     0x7f9c4bd4eeb9 - rustc_mir_transform[c290a858b6bcbc22]::check_unsafety::unsafety_check_result
  48:     0x7f9c4ac2cecc - rustc_query_impl[2b96a7f183453ea1]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[2b96a7f183453ea1]::query_impl::unsafety_check_result::dynamic_query::{closure#2}::{closure#0}, rustc_middle[a400c0366bd3264d]::query::erase::Erased<[u8; 8usize]>>
  49:     0x7f9c4ac2ceae - <rustc_query_impl[2b96a7f183453ea1]::query_impl::unsafety_check_result::dynamic_query::{closure#2} as core[afd64c1140d34ddb]::ops::function::FnOnce<(rustc_middle[a400c0366bd3264d]::ty::context::TyCtxt, rustc_span[d6219d1cb1a2cd02]::def_id::LocalDefId)>>::call_once
  50:     0x7f9c4ad49222 - rustc_query_system[f349c9744a75bab0]::query::plumbing::try_execute_query::<rustc_query_impl[2b96a7f183453ea1]::DynamicConfig<rustc_query_system[f349c9744a75bab0]::query::caches::VecCache<rustc_span[d6219d1cb1a2cd02]::def_id::LocalDefId, rustc_middle[a400c0366bd3264d]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[2b96a7f183453ea1]::plumbing::QueryCtxt, false>
  51:     0x7f9c4c4e6311 - rustc_query_impl[2b96a7f183453ea1]::query_impl::unsafety_check_result::get_query_non_incr::__rust_end_short_backtrace
  52:     0x7f9c4b3dbafc - rustc_query_impl[2b96a7f183453ea1]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[2b96a7f183453ea1]::query_impl::mir_const::dynamic_query::{closure#2}::{closure#0}, rustc_middle[a400c0366bd3264d]::query::erase::Erased<[u8; 8usize]>>
  53:     0x7f9c4b3dba4e - <rustc_query_impl[2b96a7f183453ea1]::query_impl::mir_const::dynamic_query::{closure#2} as core[afd64c1140d34ddb]::ops::function::FnOnce<(rustc_middle[a400c0366bd3264d]::ty::context::TyCtxt, rustc_span[d6219d1cb1a2cd02]::def_id::LocalDefId)>>::call_once
  54:     0x7f9c4ad49222 - rustc_query_system[f349c9744a75bab0]::query::plumbing::try_execute_query::<rustc_query_impl[2b96a7f183453ea1]::DynamicConfig<rustc_query_system[f349c9744a75bab0]::query::caches::VecCache<rustc_span[d6219d1cb1a2cd02]::def_id::LocalDefId, rustc_middle[a400c0366bd3264d]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[2b96a7f183453ea1]::plumbing::QueryCtxt, false>
  55:     0x7f9c4c4dfaa1 - rustc_query_impl[2b96a7f183453ea1]::query_impl::mir_const::get_query_non_incr::__rust_end_short_backtrace
  56:     0x7f9c4af866ff - rustc_query_impl[2b96a7f183453ea1]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[2b96a7f183453ea1]::query_impl::mir_const_qualif::dynamic_query::{closure#2}::{closure#0}, rustc_middle[a400c0366bd3264d]::query::erase::Erased<[u8; 5usize]>>
  57:     0x7f9c4bb74c54 - rustc_query_system[f349c9744a75bab0]::query::plumbing::try_execute_query::<rustc_query_impl[2b96a7f183453ea1]::DynamicConfig<rustc_query_system[f349c9744a75bab0]::query::caches::DefaultCache<rustc_span[d6219d1cb1a2cd02]::def_id::DefId, rustc_middle[a400c0366bd3264d]::query::erase::Erased<[u8; 5usize]>>, false, false, false>, rustc_query_impl[2b96a7f183453ea1]::plumbing::QueryCtxt, false>
  58:     0x7f9c4bb7491a - rustc_query_impl[2b96a7f183453ea1]::query_impl::mir_const_qualif::get_query_non_incr::__rust_end_short_backtrace
  59:     0x7f9c4ad5d5f1 - rustc_mir_transform[c290a858b6bcbc22]::mir_promoted
  60:     0x7f9c4ad5d511 - rustc_query_impl[2b96a7f183453ea1]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[2b96a7f183453ea1]::query_impl::mir_promoted::dynamic_query::{closure#2}::{closure#0}, rustc_middle[a400c0366bd3264d]::query::erase::Erased<[u8; 16usize]>>
  61:     0x7f9c4b224443 - rustc_query_system[f349c9744a75bab0]::query::plumbing::try_execute_query::<rustc_query_impl[2b96a7f183453ea1]::DynamicConfig<rustc_query_system[f349c9744a75bab0]::query::caches::VecCache<rustc_span[d6219d1cb1a2cd02]::def_id::LocalDefId, rustc_middle[a400c0366bd3264d]::query::erase::Erased<[u8; 16usize]>>, false, false, false>, rustc_query_impl[2b96a7f183453ea1]::plumbing::QueryCtxt, false>
  62:     0x7f9c4c4e06b4 - rustc_query_impl[2b96a7f183453ea1]::query_impl::mir_promoted::get_query_non_incr::__rust_end_short_backtrace
  63:     0x7f9c4b7e0e53 - rustc_borrowck[d1d386977bcd45c6]::mir_borrowck
  64:     0x7f9c4ac5ae8e - rustc_query_impl[2b96a7f183453ea1]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[2b96a7f183453ea1]::query_impl::mir_borrowck::dynamic_query::{closure#2}::{closure#0}, rustc_middle[a400c0366bd3264d]::query::erase::Erased<[u8; 8usize]>>
  65:     0x7f9c4ac5ae5e - <rustc_query_impl[2b96a7f183453ea1]::query_impl::mir_borrowck::dynamic_query::{closure#2} as core[afd64c1140d34ddb]::ops::function::FnOnce<(rustc_middle[a400c0366bd3264d]::ty::context::TyCtxt, rustc_span[d6219d1cb1a2cd02]::def_id::LocalDefId)>>::call_once
  66:     0x7f9c4ad49222 - rustc_query_system[f349c9744a75bab0]::query::plumbing::try_execute_query::<rustc_query_impl[2b96a7f183453ea1]::DynamicConfig<rustc_query_system[f349c9744a75bab0]::query::caches::VecCache<rustc_span[d6219d1cb1a2cd02]::def_id::LocalDefId, rustc_middle[a400c0366bd3264d]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[2b96a7f183453ea1]::plumbing::QueryCtxt, false>
  67:     0x7f9c4c4e9dc1 - rustc_query_impl[2b96a7f183453ea1]::query_impl::mir_borrowck::get_query_non_incr::__rust_end_short_backtrace
  68:     0x7f9c4bc81898 - rustc_mir_transform[c290a858b6bcbc22]::mir_drops_elaborated_and_const_checked
  69:     0x7f9c4ac72f6e - rustc_query_impl[2b96a7f183453ea1]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[2b96a7f183453ea1]::query_impl::mir_drops_elaborated_and_const_checked::dynamic_query::{closure#2}::{closure#0}, rustc_middle[a400c0366bd3264d]::query::erase::Erased<[u8; 8usize]>>
  70:     0x7f9c4ac72f3e - <rustc_query_impl[2b96a7f183453ea1]::query_impl::mir_drops_elaborated_and_const_checked::dynamic_query::{closure#2} as core[afd64c1140d34ddb]::ops::function::FnOnce<(rustc_middle[a400c0366bd3264d]::ty::context::TyCtxt, rustc_span[d6219d1cb1a2cd02]::def_id::LocalDefId)>>::call_once
  71:     0x7f9c4ad49222 - rustc_query_system[f349c9744a75bab0]::query::plumbing::try_execute_query::<rustc_query_impl[2b96a7f183453ea1]::DynamicConfig<rustc_query_system[f349c9744a75bab0]::query::caches::VecCache<rustc_span[d6219d1cb1a2cd02]::def_id::LocalDefId, rustc_middle[a400c0366bd3264d]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[2b96a7f183453ea1]::plumbing::QueryCtxt, false>
  72:     0x7f9c4c4dff81 - rustc_query_impl[2b96a7f183453ea1]::query_impl::mir_drops_elaborated_and_const_checked::get_query_non_incr::__rust_end_short_backtrace
  73:     0x7f9c4c14ed11 - rustc_mir_transform[c290a858b6bcbc22]::mir_for_ctfe
  74:     0x7f9c4ac98426 - rustc_query_impl[2b96a7f183453ea1]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[2b96a7f183453ea1]::query_impl::mir_for_ctfe::dynamic_query::{closure#2}::{closure#0}, rustc_middle[a400c0366bd3264d]::query::erase::Erased<[u8; 8usize]>>
  75:     0x7f9c4ac98401 - <rustc_query_impl[2b96a7f183453ea1]::query_impl::mir_for_ctfe::dynamic_query::{closure#2} as core[afd64c1140d34ddb]::ops::function::FnOnce<(rustc_middle[a400c0366bd3264d]::ty::context::TyCtxt, rustc_span[d6219d1cb1a2cd02]::def_id::DefId)>>::call_once
  76:     0x7f9c4ac53d1a - rustc_query_system[f349c9744a75bab0]::query::plumbing::try_execute_query::<rustc_query_impl[2b96a7f183453ea1]::DynamicConfig<rustc_query_system[f349c9744a75bab0]::query::caches::DefaultCache<rustc_span[d6219d1cb1a2cd02]::def_id::DefId, rustc_middle[a400c0366bd3264d]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[2b96a7f183453ea1]::plumbing::QueryCtxt, false>
  77:     0x7f9c4c4e0322 - rustc_query_impl[2b96a7f183453ea1]::query_impl::mir_for_ctfe::get_query_non_incr::__rust_end_short_backtrace
  78:     0x7f9c4b1ffb00 - <rustc_const_eval[da5ab7ed411e4025]::interpret::eval_context::InterpCx<rustc_const_eval[da5ab7ed411e4025]::const_eval::machine::CompileTimeInterpreter>>::load_mir
  79:     0x7f9c4b1013ec - rustc_const_eval[da5ab7ed411e4025]::const_eval::eval_queries::eval_to_allocation_raw_provider
  80:     0x7f9c4ac92be7 - rustc_query_impl[2b96a7f183453ea1]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[2b96a7f183453ea1]::query_impl::eval_to_allocation_raw::dynamic_query::{closure#2}::{closure#0}, rustc_middle[a400c0366bd3264d]::query::erase::Erased<[u8; 16usize]>>
  81:     0x7f9c4b78db7c - rustc_query_system[f349c9744a75bab0]::query::plumbing::try_execute_query::<rustc_query_impl[2b96a7f183453ea1]::DynamicConfig<rustc_query_system[f349c9744a75bab0]::query::caches::DefaultCache<rustc_middle[a400c0366bd3264d]::ty::ParamEnvAnd<rustc_middle[a400c0366bd3264d]::mir::interpret::GlobalId>, rustc_middle[a400c0366bd3264d]::query::erase::Erased<[u8; 16usize]>>, false, false, false>, rustc_query_impl[2b96a7f183453ea1]::plumbing::QueryCtxt, false>
  82:     0x7f9c4b78d7ef - rustc_query_impl[2b96a7f183453ea1]::query_impl::eval_to_allocation_raw::get_query_non_incr::__rust_end_short_backtrace
  83:     0x7f9c4b101994 - rustc_const_eval[da5ab7ed411e4025]::const_eval::eval_queries::eval_to_allocation_raw_provider
  84:     0x7f9c4ac92be7 - rustc_query_impl[2b96a7f183453ea1]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[2b96a7f183453ea1]::query_impl::eval_to_allocation_raw::dynamic_query::{closure#2}::{closure#0}, rustc_middle[a400c0366bd3264d]::query::erase::Erased<[u8; 16usize]>>
  85:     0x7f9c4b78db7c - rustc_query_system[f349c9744a75bab0]::query::plumbing::try_execute_query::<rustc_query_impl[2b96a7f183453ea1]::DynamicConfig<rustc_query_system[f349c9744a75bab0]::query::caches::DefaultCache<rustc_middle[a400c0366bd3264d]::ty::ParamEnvAnd<rustc_middle[a400c0366bd3264d]::mir::interpret::GlobalId>, rustc_middle[a400c0366bd3264d]::query::erase::Erased<[u8; 16usize]>>, false, false, false>, rustc_query_impl[2b96a7f183453ea1]::plumbing::QueryCtxt, false>
  86:     0x7f9c4b78d7ef - rustc_query_impl[2b96a7f183453ea1]::query_impl::eval_to_allocation_raw::get_query_non_incr::__rust_end_short_backtrace
  87:     0x7f9c4c2863e5 - rustc_const_eval[da5ab7ed411e4025]::const_eval::eval_to_valtree
  88:     0x7f9c4c286283 - <rustc_const_eval[da5ab7ed411e4025]::provide::{closure#0} as core[afd64c1140d34ddb]::ops::function::FnOnce<(rustc_middle[a400c0366bd3264d]::ty::context::TyCtxt, rustc_middle[a400c0366bd3264d]::ty::ParamEnvAnd<rustc_middle[a400c0366bd3264d]::mir::interpret::GlobalId>)>>::call_once
  89:     0x7f9c4bf1d409 - rustc_query_impl[2b96a7f183453ea1]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[2b96a7f183453ea1]::query_impl::eval_to_valtree::dynamic_query::{closure#2}::{closure#0}, rustc_middle[a400c0366bd3264d]::query::erase::Erased<[u8; 24usize]>>
  90:     0x7f9c4bf1d3b8 - <rustc_query_impl[2b96a7f183453ea1]::query_impl::eval_to_valtree::dynamic_query::{closure#2} as core[afd64c1140d34ddb]::ops::function::FnOnce<(rustc_middle[a400c0366bd3264d]::ty::context::TyCtxt, rustc_middle[a400c0366bd3264d]::ty::ParamEnvAnd<rustc_middle[a400c0366bd3264d]::mir::interpret::GlobalId>)>>::call_once
  91:     0x7f9c4c14e405 - rustc_query_system[f349c9744a75bab0]::query::plumbing::try_execute_query::<rustc_query_impl[2b96a7f183453ea1]::DynamicConfig<rustc_query_system[f349c9744a75bab0]::query::caches::DefaultCache<rustc_middle[a400c0366bd3264d]::ty::ParamEnvAnd<rustc_middle[a400c0366bd3264d]::mir::interpret::GlobalId>, rustc_middle[a400c0366bd3264d]::query::erase::Erased<[u8; 24usize]>>, false, false, false>, rustc_query_impl[2b96a7f183453ea1]::plumbing::QueryCtxt, false>
  92:     0x7f9c4c14e090 - rustc_query_impl[2b96a7f183453ea1]::query_impl::eval_to_valtree::get_query_non_incr::__rust_end_short_backtrace
  93:     0x7f9c4b260bbc - rustc_middle[a400c0366bd3264d]::query::plumbing::query_get_at::<rustc_query_system[f349c9744a75bab0]::query::caches::DefaultCache<rustc_middle[a400c0366bd3264d]::ty::ParamEnvAnd<rustc_middle[a400c0366bd3264d]::mir::interpret::GlobalId>, rustc_middle[a400c0366bd3264d]::query::erase::Erased<[u8; 24usize]>>>
  94:     0x7f9c4b260947 - <rustc_middle[a400c0366bd3264d]::ty::context::TyCtxt>::const_eval_global_id_for_typeck
  95:     0x7f9c4b2604b0 - <rustc_middle[a400c0366bd3264d]::ty::context::TyCtxt>::const_eval_resolve_for_typeck
  96:     0x7f9c4b25ed1a - <rustc_middle[a400c0366bd3264d]::ty::consts::Const>::eval
  97:     0x7f9c4b2f85a2 - <alloc[8c08642181f67925]::vec::Vec<rustc_middle[a400c0366bd3264d]::ty::Clause> as alloc[8c08642181f67925]::vec::spec_extend::SpecExtend<rustc_middle[a400c0366bd3264d]::ty::Clause, core[afd64c1140d34ddb]::iter::adapters::filter::Filter<core[afd64c1140d34ddb]::iter::adapters::map::Map<core[afd64c1140d34ddb]::iter::adapters::copied::Copied<core[afd64c1140d34ddb]::slice::iter::Iter<rustc_middle[a400c0366bd3264d]::ty::Clause>>, rustc_trait_selection[f50372471fdcdd04]::traits::normalize_param_env_or_error::{closure#0}>, <rustc_infer[5c7a2123edb7d5c]::traits::util::Elaborator<rustc_middle[a400c0366bd3264d]::ty::Clause>>::extend_deduped<core[afd64c1140d34ddb]::iter::adapters::map::Map<core[afd64c1140d34ddb]::iter::adapters::copied::Copied<core[afd64c1140d34ddb]::slice::iter::Iter<rustc_middle[a400c0366bd3264d]::ty::Clause>>, rustc_trait_selection[f50372471fdcdd04]::traits::normalize_param_env_or_error::{closure#0}>>::{closure#0}>>>::spec_extend
  98:     0x7f9c4b039440 - rustc_trait_selection[f50372471fdcdd04]::traits::normalize_param_env_or_error
  99:     0x7f9c4b0385aa - rustc_query_impl[2b96a7f183453ea1]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[2b96a7f183453ea1]::query_impl::param_env::dynamic_query::{closure#2}::{closure#0}, rustc_middle[a400c0366bd3264d]::query::erase::Erased<[u8; 8usize]>>
 100:     0x7f9c4b037e40 - <rustc_query_impl[2b96a7f183453ea1]::query_impl::param_env::dynamic_query::{closure#2} as core[afd64c1140d34ddb]::ops::function::FnOnce<(rustc_middle[a400c0366bd3264d]::ty::context::TyCtxt, rustc_span[d6219d1cb1a2cd02]::def_id::DefId)>>::call_once
 101:     0x7f9c4ac53d1a - rustc_query_system[f349c9744a75bab0]::query::plumbing::try_execute_query::<rustc_query_impl[2b96a7f183453ea1]::DynamicConfig<rustc_query_system[f349c9744a75bab0]::query::caches::DefaultCache<rustc_span[d6219d1cb1a2cd02]::def_id::DefId, rustc_middle[a400c0366bd3264d]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[2b96a7f183453ea1]::plumbing::QueryCtxt, false>
 102:     0x7f9c4c4ef801 - rustc_query_impl[2b96a7f183453ea1]::query_impl::param_env::get_query_non_incr::__rust_end_short_backtrace
 103:     0x7f9c4b0e9660 - rustc_middle[a400c0366bd3264d]::query::plumbing::query_get_at::<rustc_query_system[f349c9744a75bab0]::query::caches::DefaultCache<rustc_span[d6219d1cb1a2cd02]::def_id::DefId, rustc_middle[a400c0366bd3264d]::query::erase::Erased<[u8; 8usize]>>>
 104:     0x7f9c4b41c329 - rustc_hir_analysis[f4190e851159bd14]::check::wfcheck::check_item_fn
 105:     0x7f9c4b415714 - rustc_hir_analysis[f4190e851159bd14]::check::wfcheck::check_well_formed
 106:     0x7f9c4abfb8cc - rustc_query_impl[2b96a7f183453ea1]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[2b96a7f183453ea1]::query_impl::check_well_formed::dynamic_query::{closure#2}::{closure#0}, rustc_middle[a400c0366bd3264d]::query::erase::Erased<[u8; 0usize]>>
 107:     0x7f9c4b45961b - rustc_query_system[f349c9744a75bab0]::query::plumbing::try_execute_query::<rustc_query_impl[2b96a7f183453ea1]::DynamicConfig<rustc_query_system[f349c9744a75bab0]::query::caches::VecCache<rustc_hir[c1392d5b40313744]::hir_id::OwnerId, rustc_middle[a400c0366bd3264d]::query::erase::Erased<[u8; 0usize]>>, false, false, false>, rustc_query_impl[2b96a7f183453ea1]::plumbing::QueryCtxt, false>
 108:     0x7f9c4b459324 - rustc_query_impl[2b96a7f183453ea1]::query_impl::check_well_formed::get_query_non_incr::__rust_end_short_backtrace
 109:     0x7f9c4beac5f8 - rustc_data_structures[f18e3403d0a91f7d]::sync::par_for_each_in::<&[rustc_hir[c1392d5b40313744]::hir::ItemId], <rustc_middle[a400c0366bd3264d]::hir::ModuleItems>::par_items<rustc_hir_analysis[f4190e851159bd14]::check::wfcheck::check_mod_type_wf::{closure#0}>::{closure#0}>
 110:     0x7f9c4beac51f - rustc_hir_analysis[f4190e851159bd14]::check::wfcheck::check_mod_type_wf
 111:     0x7f9c4b7de31c - rustc_query_impl[2b96a7f183453ea1]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[2b96a7f183453ea1]::query_impl::check_mod_type_wf::dynamic_query::{closure#2}::{closure#0}, rustc_middle[a400c0366bd3264d]::query::erase::Erased<[u8; 0usize]>>
 112:     0x7f9c4b7de2fe - <rustc_query_impl[2b96a7f183453ea1]::query_impl::check_mod_type_wf::dynamic_query::{closure#2} as core[afd64c1140d34ddb]::ops::function::FnOnce<(rustc_middle[a400c0366bd3264d]::ty::context::TyCtxt, rustc_span[d6219d1cb1a2cd02]::def_id::LocalModDefId)>>::call_once
 113:     0x7f9c4bafec98 - rustc_query_system[f349c9744a75bab0]::query::plumbing::try_execute_query::<rustc_query_impl[2b96a7f183453ea1]::DynamicConfig<rustc_query_system[f349c9744a75bab0]::query::caches::DefaultCache<rustc_span[d6219d1cb1a2cd02]::def_id::LocalModDefId, rustc_middle[a400c0366bd3264d]::query::erase::Erased<[u8; 0usize]>>, false, false, false>, rustc_query_impl[2b96a7f183453ea1]::plumbing::QueryCtxt, false>
 114:     0x7f9c4c4e9284 - rustc_query_impl[2b96a7f183453ea1]::query_impl::check_mod_type_wf::get_query_non_incr::__rust_end_short_backtrace
 115:     0x7f9c4bdbe89e - rustc_data_structures[f18e3403d0a91f7d]::sync::par_for_each_in::<&[rustc_hir[c1392d5b40313744]::hir_id::OwnerId], <rustc_middle[a400c0366bd3264d]::hir::map::Map>::par_for_each_module<rustc_hir_analysis[f4190e851159bd14]::check_crate::{closure#5}::{closure#0}::{closure#0}>::{closure#0}>
 116:     0x7f9c4bdbe697 - <rustc_session[bc10157bc0a34fd]::session::Session>::track_errors::<rustc_hir_analysis[f4190e851159bd14]::check_crate::{closure#5}, ()>
 117:     0x7f9c4bdbd278 - rustc_hir_analysis[f4190e851159bd14]::check_crate
 118:     0x7f9c4bdbc172 - rustc_interface[32fae85809e14be6]::passes::analysis
 119:     0x7f9c4bfc51ca - rustc_query_impl[2b96a7f183453ea1]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[2b96a7f183453ea1]::query_impl::analysis::dynamic_query::{closure#2}::{closure#0}, rustc_middle[a400c0366bd3264d]::query::erase::Erased<[u8; 1usize]>>
 120:     0x7f9c4bfc51b9 - <rustc_query_impl[2b96a7f183453ea1]::query_impl::analysis::dynamic_query::{closure#2} as core[afd64c1140d34ddb]::ops::function::FnOnce<(rustc_middle[a400c0366bd3264d]::ty::context::TyCtxt, ())>>::call_once
 121:     0x7f9c4c27e394 - rustc_query_system[f349c9744a75bab0]::query::plumbing::try_execute_query::<rustc_query_impl[2b96a7f183453ea1]::DynamicConfig<rustc_query_system[f349c9744a75bab0]::query::caches::SingleCache<rustc_middle[a400c0366bd3264d]::query::erase::Erased<[u8; 1usize]>>, false, false, false>, rustc_query_impl[2b96a7f183453ea1]::plumbing::QueryCtxt, false>
 122:     0x7f9c4c27e109 - rustc_query_impl[2b96a7f183453ea1]::query_impl::analysis::get_query_non_incr::__rust_end_short_backtrace
 123:     0x7f9c4c1793f3 - <rustc_middle[a400c0366bd3264d]::ty::context::GlobalCtxt>::enter::<rustc_driver_impl[bd3214347f952018]::run_compiler::{closure#1}::{closure#2}::{closure#6}, core[afd64c1140d34ddb]::result::Result<(), rustc_span[d6219d1cb1a2cd02]::ErrorGuaranteed>>
 124:     0x7f9c4c07d4ec - <rustc_interface[32fae85809e14be6]::interface::Compiler>::enter::<rustc_driver_impl[bd3214347f952018]::run_compiler::{closure#1}::{closure#2}, core[afd64c1140d34ddb]::result::Result<core[afd64c1140d34ddb]::option::Option<rustc_interface[32fae85809e14be6]::queries::Linker>, rustc_span[d6219d1cb1a2cd02]::ErrorGuaranteed>>
 125:     0x7f9c4c074ce8 - std[15568f5d89cfcf8b]::sys_common::backtrace::__rust_begin_short_backtrace::<rustc_interface[32fae85809e14be6]::util::run_in_thread_with_globals<rustc_interface[32fae85809e14be6]::interface::run_compiler<core[afd64c1140d34ddb]::result::Result<(), rustc_span[d6219d1cb1a2cd02]::ErrorGuaranteed>, rustc_driver_impl[bd3214347f952018]::run_compiler::{closure#1}>::{closure#0}, core[afd64c1140d34ddb]::result::Result<(), rustc_span[d6219d1cb1a2cd02]::ErrorGuaranteed>>::{closure#0}::{closure#0}, core[afd64c1140d34ddb]::result::Result<(), rustc_span[d6219d1cb1a2cd02]::ErrorGuaranteed>>
 126:     0x7f9c4c07445e - <<std[15568f5d89cfcf8b]::thread::Builder>::spawn_unchecked_<rustc_interface[32fae85809e14be6]::util::run_in_thread_with_globals<rustc_interface[32fae85809e14be6]::interface::run_compiler<core[afd64c1140d34ddb]::result::Result<(), rustc_span[d6219d1cb1a2cd02]::ErrorGuaranteed>, rustc_driver_impl[bd3214347f952018]::run_compiler::{closure#1}>::{closure#0}, core[afd64c1140d34ddb]::result::Result<(), rustc_span[d6219d1cb1a2cd02]::ErrorGuaranteed>>::{closure#0}::{closure#0}, core[afd64c1140d34ddb]::result::Result<(), rustc_span[d6219d1cb1a2cd02]::ErrorGuaranteed>>::{closure#1} as core[afd64c1140d34ddb]::ops::function::FnOnce<()>>::call_once::{shim:vtable#0}
 127:     0x7f9c4dcea015 - <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once::he46350f6bfcc0e8a
                               at /rustc/9f5fc1bd443f59583e7af0d94d289f95fe1e20c4/library/alloc/src/boxed.rs:2007:9
 128:     0x7f9c4dcea015 - <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once::h4038c4c7fc344b18
                               at /rustc/9f5fc1bd443f59583e7af0d94d289f95fe1e20c4/library/alloc/src/boxed.rs:2007:9
 129:     0x7f9c4dcea015 - std::sys::unix::thread::Thread::new::thread_start::h842f0d826bbbe958
                               at /rustc/9f5fc1bd443f59583e7af0d94d289f95fe1e20c4/library/std/src/sys/unix/thread.rs:108:17
 130:     0x7f9c4968c9eb - <unknown>
 131:     0x7f9c49710dfc - <unknown>
 132:                0x0 - <unknown>

error: the compiler unexpectedly panicked. this is a bug.

note: we would appreciate a bug report: https://github.com/rust-lang/rust/issues/new?labels=C-bug%2C+I-ICE%2C+T-compiler&template=ice.md

note: please attach the file at `/tmp/ff/icemaker_reduced/rustc-ice-2023-09-03T17:21:42.182195429Z-2372159.txt` to your bug report

query stack during panic:
#0 [typeck] type-checking `assert::is_maybe_transmutable::{constant#0}`
#1 [thir_body] building THIR for `assert::is_maybe_transmutable::{constant#0}`
#2 [check_match] match-checking `assert::is_maybe_transmutable::{constant#0}`
#3 [mir_built] building MIR for `assert::is_maybe_transmutable::{constant#0}`
#4 [unsafety_check_result] unsafety-checking `assert::is_maybe_transmutable::{constant#0}`
#5 [mir_const] preparing `assert::is_maybe_transmutable::{constant#0}` for borrow checking
#6 [mir_const_qualif] const checking `assert::is_maybe_transmutable::{constant#0}`
#7 [mir_promoted] promoting constants in MIR for `assert::is_maybe_transmutable::{constant#0}`
#8 [mir_borrowck] borrow-checking `assert::is_maybe_transmutable::{constant#0}`
#9 [mir_drops_elaborated_and_const_checked] elaborating drops for `assert::is_maybe_transmutable::{constant#0}`
#10 [mir_for_ctfe] caching mir of `assert::is_maybe_transmutable::{constant#0}` for CTFE
#11 [eval_to_allocation_raw] const-evaluating + checking `assert::is_maybe_transmutable::{constant#0}`
#12 [eval_to_allocation_raw] const-evaluating + checking `assert::is_maybe_transmutable::{constant#0}`
#13 [eval_to_valtree] evaluating type-level constant
#14 [param_env] computing normalized predicates of `assert::is_maybe_transmutable`
#15 [check_well_formed] checking that `assert::is_maybe_transmutable` is well-formed
#16 [check_mod_type_wf] checking that types are well-formed in module `assert`
#17 [analysis] running analysis passes on this crate
end of query stack
error: aborting due to 13 previous errors

Some errors have detailed explanations: E0061, E0229, E0412, E0601, E0658.
For more information about an error, try `rustc --explain E0061`.
```

</p>
</details>

--- a/tests/ui/typeck/escaping_bound_vars.rs
+++ b/tests/ui/typeck/escaping_bound_vars.rs
@@ -1,0 +1,16 @@
+// Test for issues/115517 which is fixed by pull/115486
+// This should not ice
+trait Test<const C: usize> {}
+
+trait Elide<T> {
+    fn call();
+}
+
+pub fn test()
+where
+    (): Test<{ 1 + (<() as Elide(&())>::call) }>,
+    //~^ ERROR cannot capture late-bound lifetime in constant
+{
+}
+
+fn main() {}

--- a/tests/ui/typeck/escaping_bound_vars.stderr
+++ b/tests/ui/typeck/escaping_bound_vars.stderr
@@ -1,0 +1,10 @@
+error: cannot capture late-bound lifetime in constant
+  --> $DIR/escaping_bound_vars.rs:11:35
+   |
+LL |     (): Test<{ 1 + (<() as Elide(&())>::call) }>,
+   |                                  -^
+   |                                  |
+   |                                  lifetime defined here
+
+error: aborting due to previous error
+

